commands/provider/service_deployer.go

```diff
@@ -53,11 +53,12 @@ func (p *ServiceProvider) ServiceDeploy(path string, statuses chan DeployStatus)
 	if govalidator.IsURL(path) {
 		if err := stream.Send(&coreapi.DeployServiceRequest{
 			Value: &coreapi.DeployServiceRequest_Url{Url: path},
+			Env:   env,
```

Comment:
We don't have an API guidelines/flow.

I send env with a first message always and in the server I expected env to be in a first message.
Reason? With such approach 
- you will have to send only 1 message (in case of deploy from url) and not 2 
- on first message api can decide if this will be url deploy or tarball
- if cli will send first message with env only then we need to read first message, check if it has url/chunk - if not then read second message and then decided.

With this approach, the first message always contains env + chunk/url. Ofc env may appear in any message , but only first one is valid.

Reply:
it's ambiguous to send env variables with url when url is the option and in a separete message when chunk is the option. you can send the env variable by it's own as the first message of the stream and implement it under oneOf https://github.com/mesg-foundation/core/pull/641#discussion_r241668096. this is our guideline so far.

Reply:
I suggest to send `env` in the first message with `URL` or `Chunk`.
It will make the reading from the core easier.

Reply:
solved by https://github.com/mesg-foundation/core/pull/654/commits/59b574fec3d569643e0f8c114139f1f8f96e685c:

deploy api now has 3 fields(inside onOf):
* options
* url
* chunk

these fields should be sent to service in the same order.
* options must be sent as 1th even if it's empty.
* url or chunks must be sent as 2nd (if chunks chosen, client should close the stream after all chunks sent).

this flow is much easier to implement by the server and gRPC clients(our consumers).